app-main/src/main/kotlin/kr/flab/wiki/app/components/authentication/LoginUserService.kt

```diff
@@ -0,0 +1,30 @@
+package kr.flab.wiki.app.components.authentication
+
+import kr.flab.wiki.app.api.user.request.LoginRequest
+import kr.flab.wiki.app.api.user.response.LoginResponse
+import kr.flab.wiki.app.utils.JwtUtils
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Service
+
+@Service
+class LoginUserService(
+    private val authenticationManager: AuthenticationManager,
+) {
+
+    fun login(loginRequest: LoginRequest): LoginResponse? {
+        val authentication: Authentication = authenticationManager.authenticate(
+            UsernamePasswordAuthenticationToken(
+                loginRequest.email,
+                loginRequest.password
+            )
+        )
+        if (authentication.principal == null) {
```

Comment:
spring security 에서 말하는 *principal* 이란 어떤 의미인가요?

Reply:
현재 로그인한 유저의 정보입니다.

Reply:
```
/**
 * This interface represents the abstract notion of a principal, which
 * can be used to represent any entity, such as an individual, a
 * corporation, and a login id.
 *
 * @see java.security.cert.X509Certificate
 *
 * @author Li Gong
 * @since 1.1
 */
```

이라고 합니다. `java.security.Principal` 소스 코드 한번 읽어보세요.